datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1946,6 +1950,47 @@ mod tests {
     };
     use datafusion_common::scalar::ScalarValue;
 
+    #[test]
+    fn test_align_array_dimensions() {
+        let array1d_1 =
+            Arc::new(ListArray::from_iter_primitive::<Int64Type, _, _>(vec![
+                Some(vec![Some(1), Some(2), Some(3)]),
+                Some(vec![Some(4), Some(5)]),
+            ]));
+        let array1d_2 =
+            Arc::new(ListArray::from_iter_primitive::<Int64Type, _, _>(vec![
+                Some(vec![Some(6), Some(7), Some(8)]),
+            ]));
+
+        let array2d_1 = Arc::new(wrap_into_list_array(array1d_1.clone())) as ArrayRef;
+        let array2d_2 = Arc::new(wrap_into_list_array(array1d_2.clone())) as ArrayRef;
+
+        let res =
+            align_array_dimensions(vec![array1d_1.to_owned(), array2d_2.to_owned()])
+                .expect("should not error");
```

Comment:
Please add more descriptive text in `expect`

Reply:
Could I change them all to unwrap for test?

Reply:
I think we could only unwrap in tests based on this [influxdata/influxdb@main/docs/style_guide.md#dont-return-result-from-test-functions](https://github.com/influxdata/influxdb/blob/main/docs/style_guide.md?rgh-link-date=2023-10-26T16%3A42%3A50Z#dont-return-result-from-test-functions)